File: depend/libcmb/perl/test4.pl
#!/usr/bin/env perl
use strict;
use warnings;
use Cmb;

my @items = (1..10000);
my $choose = 2;
my $nitems = $#items + 1;

our $DPV;
my $total = qx/cmb -k $choose -T $nitems/;
chomp $total;
open $DPV, "|-", "dpv -l $total:-" or die "dpv: $!";

my $cmb = new Cmb { size_min => $choose, size_max => $choose };
$cmb->cmb_callback($nitems, \@items, sub { print $DPV "@_\n" });
close $DPV;

File: depend/libcmb/python/test4.py
#!/usr/bin/env python
from cmb import *
import signal
import subprocess
import sys

def interrupt(sig, frame):
    sys.exit(0)

signal.signal(signal.SIGINT, interrupt)

items = range(0, 1000)
choose = 2
nitems = len(items)

config = CMB(size_min = choose, size_max = choose)
total = cmb_count(config, nitems)
dpv = subprocess.Popen(["dpv", "-l", "%u:perl" % total],
    stdin = subprocess.PIPE)

def afunc(items):
    global dpv
    dpv.stdin.write(" ".join(items) + "\n")
    return 0
cmb_callback(config, nitems, items, afunc)
